libs/vkd3d/utils.c

```diff
@@ -153,6 +153,239 @@ static const struct vkd3d_format vkd3d_depth_stencil_formats[] =
+    {DXGI_FORMAT_R16_SINT,
+            {DXGI_FORMAT_R16_UINT, DXGI_FORMAT_R16_UNORM, DXGI_FORMAT_R16_SNORM},
+            DXGI_FORMAT_R16_UINT},
+    {DXGI_FORMAT_R16_UNORM,
+            {DXGI_FORMAT_R16_UINT, DXGI_FORMAT_R16_SINT},
+            DXGI_FORMAT_R16_UINT},
+    {DXGI_FORMAT_R16_SNORM,
+            {DXGI_FORMAT_R16_UINT, DXGI_FORMAT_R16_SINT},
+            DXGI_FORMAT_R16_UINT},
+
+    {DXGI_FORMAT_R8_TYPELESS,
+            {DXGI_FORMAT_R8_UINT, DXGI_FORMAT_R8_SINT, DXGI_FORMAT_R8_UNORM, DXGI_FORMAT_R8_SNORM, DXGI_FORMAT_A8_UNORM},
+            DXGI_FORMAT_R8_UINT},
+    {DXGI_FORMAT_R8_UINT,
+            {DXGI_FORMAT_R8_SINT, DXGI_FORMAT_R8_UNORM, DXGI_FORMAT_R8_SNORM, DXGI_FORMAT_A8_UNORM},
+            DXGI_FORMAT_R8_UINT},
+    {DXGI_FORMAT_R8_SINT,
+            {DXGI_FORMAT_R8_UINT, DXGI_FORMAT_R8_UNORM, DXGI_FORMAT_R8_SNORM, DXGI_FORMAT_A8_UNORM},
+            DXGI_FORMAT_R8_UINT},
+    {DXGI_FORMAT_R8_UNORM,
+            {DXGI_FORMAT_R8_UINT, DXGI_FORMAT_R8_SINT, DXGI_FORMAT_A8_UNORM},
+            DXGI_FORMAT_R8_UINT},
+    {DXGI_FORMAT_R8_SNORM,
+            {DXGI_FORMAT_R8_UINT, DXGI_FORMAT_R8_SINT},
+            DXGI_FORMAT_R8_UINT},
+    {DXGI_FORMAT_A8_UNORM,
+            {DXGI_FORMAT_R8_UINT, DXGI_FORMAT_R8_SINT, DXGI_FORMAT_R8_UNORM},
+            DXGI_FORMAT_R8_UINT},
+
+    {DXGI_FORMAT_B8G8R8A8_TYPELESS,
+            {DXGI_FORMAT_B8G8R8A8_UNORM, DXGI_FORMAT_B8G8R8A8_UNORM_SRGB},
+            DXGI_FORMAT_R8G8B8A8_UINT},
+    {DXGI_FORMAT_B8G8R8A8_UNORM,
+            {DXGI_FORMAT_B8G8R8A8_UNORM_SRGB},
+            DXGI_FORMAT_R8G8B8A8_UINT},
+    {DXGI_FORMAT_B8G8R8A8_UNORM_SRGB,
+            {DXGI_FORMAT_B8G8R8A8_UNORM},
+            DXGI_FORMAT_R8G8B8A8_UINT},
+
+    {DXGI_FORMAT_B8G8R8X8_TYPELESS,
+            {DXGI_FORMAT_B8G8R8X8_UNORM, DXGI_FORMAT_B8G8R8X8_UNORM_SRGB},
+            DXGI_FORMAT_R8G8B8A8_UINT},
+    {DXGI_FORMAT_B8G8R8X8_UNORM,
+            {DXGI_FORMAT_B8G8R8X8_UNORM_SRGB},
+            DXGI_FORMAT_R8G8B8A8_UINT},
+    {DXGI_FORMAT_B8G8R8X8_UNORM_SRGB,
+            {DXGI_FORMAT_B8G8R8X8_UNORM},
+            DXGI_FORMAT_R8G8B8A8_UINT},
+
+    {DXGI_FORMAT_BC1_TYPELESS,
+            {DXGI_FORMAT_BC1_UNORM, DXGI_FORMAT_BC1_UNORM_SRGB}},
+    {DXGI_FORMAT_BC1_UNORM,
+            {DXGI_FORMAT_BC1_UNORM_SRGB}},
+    {DXGI_FORMAT_BC1_UNORM_SRGB,
+            {DXGI_FORMAT_BC1_UNORM}},
+
+    {DXGI_FORMAT_BC2_TYPELESS,
+            {DXGI_FORMAT_BC2_UNORM, DXGI_FORMAT_BC2_UNORM_SRGB}},
+    {DXGI_FORMAT_BC2_UNORM,
+            {DXGI_FORMAT_BC2_UNORM_SRGB}},
+    {DXGI_FORMAT_BC2_UNORM_SRGB,
+            {DXGI_FORMAT_BC2_UNORM}},
+
+    {DXGI_FORMAT_BC3_TYPELESS,
+            {DXGI_FORMAT_BC3_UNORM, DXGI_FORMAT_BC3_UNORM_SRGB}},
+    {DXGI_FORMAT_BC3_UNORM,
+            {DXGI_FORMAT_BC3_UNORM_SRGB}},
+    {DXGI_FORMAT_BC3_UNORM_SRGB,
+            {DXGI_FORMAT_BC3_UNORM}},
+
+    {DXGI_FORMAT_BC4_TYPELESS,
+            {DXGI_FORMAT_BC4_UNORM, DXGI_FORMAT_BC4_SNORM}},
+    {DXGI_FORMAT_BC5_TYPELESS,
```

Comment:
Missed BC4_UNORM, BC4_SNORM? Same with BC5.

Reply:
Is this allowed under the no SNORM <-> UNORM rule?

Reply:
Oh, I see. It's not I think.

Reply:
I figure if a format is not present in this list, it's considered non-mutable?

Reply:
Yes, however we might still make images mutable regardless if UAV usage is enabled (under the uint/R32UI rule).